### PR TITLE
Support strong mode

### PR DESCRIPTION
**Pull request by lukechurch**
Upgrade to the latest version of the VM, add an optional argument to support strong mode.

### CODE REVIEW

**Comment by lukechurch** [comment]
Build issues are cosmetic, I'll get a fix shortly: https://github.com/dart-lang/dart-services/issues/269

File: lib/src/common_server.dart [hunk 1]
```diff
@@ -275,24 +279,28 @@ class CommonServer {
...
       throw new BadRequestError('Missing parameter: \'sources\'');
     }
     Stopwatch watch = new Stopwatch()..start();
     String sourcesJson = new JsonEncoder().convert(sources);
-    srcRequestRecorder.record("ANALYZE-v1", sourcesJson);
+    srcRequestRecorder.record("ANALYZE-v2-$strongMode", sourcesJson);
     _logger.info("About to ANALYZE-v1: ${_hashSource(sourcesJson)}");
+
+    // Select the right analyzer
+    Analyzer selectedAnalyzer = strongMode ? strongModeAnalyzer : analyzer;
```
Comment (devoncarew) [inline lib/src/common_server.dart]: The `strongMode` param is coming from the request - you may want to guard against it being null here. Something like `strongMode ??= false` before this line.

Comment (lukechurch) [inline lib/src/common_server.dart]: It'll default to false if it is null at the request handler, but I've no objection to putting in the extra-defensiveness.

Comment (devoncarew) [inline lib/src/common_server.dart]: I'm you're sure, then there's no need. I was worried tracing through the logic that we could be passing a `null` value into a named param, which would override the default of `strongMode: false`.

Comment (lukechurch) [inline lib/src/common_server.dart]: I prefer the check - added, thanks for the catch.

**Pull request merged**